### PR TITLE
Add MIDI record, support for ALSA MIDI in Linux

### PR DESCRIPTION
**Pull request by cyclopsian**
Makes it so notes from MIDI devices actually add the notes to the pattern instead of just playing them. Please test on other platforms to ensure this didn't break anything.

### CODE REVIEW

**Comment by kometbomb** [comment]
Very cool! Need to check this before merging.

File: src/main.c [hunk 1]
```diff
@@ -477,6 +477,12 @@ int main(int argc, char **argv)
 				case MSG_NOTEOFF:
 				case MSG_PROGRAMCHANGE:
 					note_event(&e);
+				case MSG_CLOCK:
```
Comment (rofl0r) [inline src/main.c]: `/* fall-through ? */`

Comment (cyclopsian) [reply src/main.c]: Good catch, added a break statement there

**Comment by kometbomb** [comment]
I am able to enter notes in the pattern on Windows. If you didn't add any functionality that is not ported to the Windows side of the MIDI code, I think this is ready to go.

**Comment by cyclopsian** [comment]
@kometbomb There is no new functionality that should affect the windows side of things, I tried not to mess with that.

**Pull request merged**